### PR TITLE
Redesign-profile-page

### LINKED ISSUE #315: Redesign the Profile Page
## Issue
- The Profile route of the site is of no use
- It has a dumb form which cannot be used to update any data

## Points to Improve on
- Only preview the User Details of the logged in user
Reason : User can't update them through my-site, it will happen through the identity service they will deploy (for developers) and for others (they can maybe edit through my-site) - Depends on their role they have
- Maybe show their GitHub Profile, if they have
- Preview the Profile Picture and fix the UI of the Upload photo

## Current Profile Page
https://my.realdevsquad.com/profile

image
image

## Figma Design
https://www.figma.com/file/KdayucUZf6d6bZkOSogdY8/Profile-Page?node-id=0%3A1&t=lzPfvJYu9ni8eckT-1

### PR DESCRIPTION
**Pull request by bhtibrewal**
## Issue
Closes #315

## Visual Changes
image
image
image

image
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,8 +29,9 @@
   },
   "dependencies": {
     "dotenv": "^16.0.1",
-    "mixpanel-browser": "^2.43.0",
+    "ember-cli-app-version": "^5.0.0",
```
Comment (TanishqSingla) [inline package.json:32]: I think we should have a separate PR to update the CLI to separate the concerns. Is that possible?

File: app/styles/profile.css [hunk 2]
```diff
@@ -1,14 +1,74 @@
-.profile-container {
+.profile-background {
+  background: var(--nav--bg);
+  height: 50vh;
+  /* position: absolute;
+  left: 0px;
+  width: 100%; */
```
Comment (TanishqSingla) [inline app/styles/profile.css:6]: can we remove this comment?

**Comment by bhtibrewal** [comment]
Update: functionality and UI changes done and verified with @rohan09-raj working on adding tests for the profile components

File: app/components/image-cropper.js [hunk 3]
```diff
@@ -22,7 +22,7 @@ export default class ImageCropperComponent extends Component {
       zoomOnWheel: false,
       rotatable: false,
       toggleDragModeOnDblclick: false,
-      preview: '.image-cropper-preview',
+      // preview: '.image-cropper-preview',
```
Comment (rohan09-raj) [inline app/components/image-cropper.js:25]: Can we please remove the commented code?

File: app/components/upload-image.hbs [hunk 4]
```diff
@@ -1,59 +1,77 @@
...
+{{#if this.isImageSelected}}
+  <h1>
+    Crop Selected Image
+  </h1>
+
+  <ImageCropper
+    @image={{this.image}}
+    @setImageCoordinates={{this.setImageCoordinates}}
+  />
+  <div class=''>
```
Comment (rohan09-raj) [inline app/components/upload-image.hbs:10]: Can we remove the empty class div?

File: app/styles/profile.css [hunk 5]
```diff
@@ -1,14 +1,77 @@
-.profile-container {
+.profile-background {
+  background: var(--nav--bg);
+  height: 50vh;
+}
+.profile-form {
+  margin-top: calc(-45vh);
+  background-color: #ffffff;
```
Comment (rohan09-raj) [inline app/styles/profile.css:7]: Use the color variables here

File: package.json [hunk 6]
```diff
@@ -30,8 +30,9 @@
...
```
Comment (rohan09-raj) [inline package.json:33]: Can we please move this dependency back to `devDependencies`

File: tests/integration/components/image-cropper-test.js [hunk 7]
```diff
@@ -1,26 +1,17 @@
...
+import { module, test } from 'qunit';
 import { setupRenderingTest } from 'ember-qunit';
 import { render } from '@ember/test-helpers';
 import { hbs } from 'ember-cli-htmlbars';
 
 module('Integration | Component | image-cropper', function (hooks) {
   setupRenderingTest(hooks);
 
-  skip('it renders', async function (assert) {
+  test('it renders', async function (assert) {
```
Comment (rohan09-raj) [inline tests/integration/components/image-cropper-test.js:9]: Can we write tests for the image cropper as well ?

File: tests/integration/components/upload-image-test.js [hunk 8]
```diff
@@ -1,23 +1,55 @@
...
+    <UploadImage @uploadUrl={{this.uploadUrl}}  @formKeyName = {{this.formDataKeyName}}/>
+    `);
+    const button = this.element.querySelector('.image-form__button');
+
+    assert.dom('.drop-area').exists().hasNoClass('.drop-area__highlight');
+
+    // Test handling of file input change event
+    await triggerEvent('input[type="file"]', 'change', { files: [file] });
+
+    assert.dom('.drop-area').doesNotExist();
```
Comment (rohan09-raj) [inline tests/integration/components/upload-image-test.js:25]: Can we please use `data-test-selectors` here ? As the sole purpose of `data-test-selectors` is to grab elements for testing only while classes may be conflicting due to being used with various elements

File: tests/unit/routes/profile-test.js [hunk 9]
```diff
@@ -8,4 +9,15 @@ module('Unit | Route | profile', function (hooks) {
     let route = this.owner.lookup('route:profile');
     assert.ok(route);
   });
+
+  test('test modal close on back button click', async (assert) => {
+    await visit('/profile');
+
+    assert.dom('[data-test-btn="edit"]').exists();
+    click('.edit-btn');
```
Comment (rohan09-raj) [inline tests/unit/routes/profile-test.js:17]: can we please use data-test-selector here as well?

**Review by rohan09-raj — CHANGES_REQUESTED** [review]
- The API call to update image is failing and giving the `status code 500` in my local environment. Please have a look at it.
image

- The modal is not responsive, can we please add media queries to make it responsive?
 
- There are lot of unnecessary code formatting changes. Please make sure to not push those, we can separately take care of formatting the files which has no effect in this feature. You can disable your code formatters `on save` or can remove from `settings.json` as well. For now, leave it as it is.

- I have requested few changes, please address them

- Also make sure when you undo the changes of package.json, revert the changes of yarn.lock as well.

File: app/components/image-cropper.js (hunk 3 above)
Comment (bhtibrewal) [reply app/components/image-cropper.js:25]: done

File: app/components/upload-image.hbs (hunk 4 above)
Comment (bhtibrewal) [reply app/components/upload-image.hbs:10]: done

File: tests/integration/components/image-cropper-test.js (hunk 7 above)
Comment (bhtibrewal) [reply tests/integration/components/image-cropper-test.js:9]: I tried but I really can't understand what to write, also there is no documentation for testing the cropper js library

Comment (rohan09-raj) [reply tests/integration/components/image-cropper-test.js:9]: Ok got it

File: app/styles/profile.css
Comment (rohan09-raj) [inline app/styles/profile.css:1]: Can we please fix this, the dropdown should be on the top layer?
image

Comment (rohan09-raj) [reply app/styles/profile.css:1]: Also, can we please make the modal responsive
image

The complete modal is not visible here
image

Comment (rohan09-raj) [reply app/styles/profile.css:1]: There is extra space left at the bottom of the card. Can we make a whose height adjusts according to the elements it contain.
image

Comment (rohan09-raj) [reply app/styles/profile.css:1]: For this web view also, we can better margins and paddings
image

Comment (rohan09-raj) [reply app/styles/profile.css:1]: Can we placeholders for text and image as well?

Comment (bhtibrewal) [reply app/styles/profile.css:1]: image
placeholder text and image

Comment (bhtibrewal) [reply app/styles/profile.css:1]: > Can we please fix this, the dropdown should be on the top layer? image

image

**Pull request merged**